Client/Frontend/Autofill/CreditCard/ViewComponents/SensitiveHostingController.swift

```diff
@@ -0,0 +1,82 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import SwiftUI
+
+
+/// The `SensitiveHostingController` blurs the nested view on backgrounding and asks to authenticate before foregrounding.
+///
+/// "Sensitive" refers to a screen with sensitive user data. Typically, this data is hidden behind `LocalAuthentication` and
+/// a user must authenticate each time.
+class SensitiveHostingController<Content>: UIHostingController<Content> where Content: View {
+    private var blurredOverlay: UIImageView?
+    private var isAuthenticated = false
+    private var willEnterForegroundNotificationObserver: NSObjectProtocol?
+    private var didEnterBackgroundNotificationObserver: NSObjectProtocol?
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        willEnterForegroundNotificationObserver = NotificationCenter.default.addObserver(forName: UIApplication.willEnterForegroundNotification, object: nil, queue: .main) { [self] notification in
+            if !isAuthenticated {
+                AppAuthenticator.authenticateWithDeviceOwnerAuthentication { [self] result in
+                    switch result {
+                    case .success:
+                        isAuthenticated = false
```

Comment:
Why is `isAuthenticated` false when the authentication succeeds?

Reply:
This caught me too but I kept it pretty much the same as what we have in SensitiveViewController, since that behavior is giving what we expected from the Passwords screen. I can look into that.

Reply:
My understanding from SensitiveViewController is that default authentication is false because in order for us to go into the view (Ex. logins) you start with false authentication. Then you are asked by the system to authenticate via biometrics. 

This should be the same for Edit View in `View Only` mode. Basically, the flow is

- User taps on a credit card info cell
- App asks for biometrics via isAuthenticated and checks for biometrics 
- User backgrounds the app we set isAuthenticated to false as it should ask for authentication again or check it based on `authenticateWithDeviceOwnerAuthentication method